src/views/Pipeline.vue

```diff
@@ -117,6 +117,48 @@ export default defineComponent({
         event.target.complete();
       })
     },
+    async skipJob (job: any) {
+      const alert = await alertController
+        .create({
+          header: this.$t('Skip job'),
+          message: this.$t('Skipping will run this job at the next occurance based on the temporal expression.'),
+          buttons: [
+            {
+              text: this.$t('Dont skip'),
+              role: 'cancel',
+            },
+            {
+              text: this.$t('Skip'),
+              handler: () => {
+                if(this.temporalExpr(job.tempExprId).integer1 === 12){
+                  const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus({minutes: this.temporalExpr(job.tempExprId).integer2});
+                  const timeDiff = job.runTime - DateTime.local().toMillis()
+                  job.runTime = job.runTime + time - timeDiff
+                }
+                else if (this.temporalExpr(job.tempExprId).integer1 === 10){
+                  const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus({hours: this.temporalExpr(job.tempExprId).integer2});
+                  const timeDiff = job.runTime - DateTime.local().toMillis()
+                  job.runTime = job.runTime + time - timeDiff
+                }
+                else if (this.temporalExpr(job.tempExprId).integer1 === 5){
+                  const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus({days: this.temporalExpr(job.tempExprId).integer2});  
+                  const timeDiff = job.runTime - DateTime.local().toMillis()
+                  job.runTime = job.runTime + time - timeDiff
```

Comment:
We can optimize this code by writing these three lines outside of the if-else and initializing a variable on the basis of the condition, which will be further used in the `plus` method.
```
let skipTime = '' // outside if-else

skipTime = {days: this.temporalExpr(job.tempExprId).integer2} // on the basis of condition
                  
// outside if-else block
const time =  DateTime.fromMillis(job.runTime).diff(DateTime.local()).plus(skipTime);  
const timeDiff = job.runTime - DateTime.local().toMillis()
job.runTime = job.runTime + time - timeDiff
```

Reply:
Improved the code.

Reply:
Thank You.